src/target/llvm/codegen_hexagon.cc

```diff
@@ -706,12 +706,42 @@ runtime::Module BuildHexagon(IRModule mod, Target target) {
   std::unique_ptr<llvm::TargetMachine> tm = GetLLVMTargetMachine(target);
   std::unique_ptr<llvm::LLVMContext> ctx(new llvm::LLVMContext());
   std::unique_ptr<CodeGenHexagon> cg(new CodeGenHexagon());
-  cg->Init("TVMHexagonModule", tm.get(), ctx.get(), false, false, false);
+
+  std::vector<PrimFunc> funcs;
+  std::string entry_func;
+  Map<String, LinkedParam> linked_params;
+  bool found_linked_params = false;
+  bool could_have_linked_params = target->GetAttr<Bool>("link-params").value_or(Bool(false));
+
   for (auto kv : mod->functions) {
     ICHECK(kv.second->IsInstance<PrimFuncNode>()) << "Can only lower IR Module with PrimFuncs";
+    if (could_have_linked_params &&
+        kv.first->name_hint == ::tvm::runtime::symbol::tvm_lookup_linked_param) {
+      Map<String, ObjectRef> attrs_dict = Downcast<Map<String, ObjectRef>>(kv.second->attrs->dict);
+      CHECK(attrs_dict.find(::tvm::tir::attr::kLinkedParams) != attrs_dict.end())
+          << "no " << ::tvm::tir::attr::kLinkedParams << " attribute found!";
+      linked_params =
```

Comment:
Why are we replacing the map here (as opposed to setting it) ?

Reply:
What do you mean by "replacing" the map?  `Map` is an `ObjectRef`, so we're not really copying anything here.

Reply:
I thought the reason for declaring the Map outside of the scope of loop is to set it -- hence the question.

Another way to ask the same thing : any reason to declare the Map outside of the for loop ?

Reply:
It's used after the loop.  The map holds the parameters that we pass to `LinkParameters`.

Reply:
Hmmm, it seems this work because it is only set once in the loop. 
Any reason not to codegen it here (rather than doing it down there) ?